### PR TITLE
Azure/000

### PR DESCRIPTION
**Pull request by WillBeebe**

### CODE REVIEW

File: massdriver-application-azure-app-service/subnet.tf [hunk 1]
```diff
@@ -50,11 +49,11 @@ resource "utility_available_cidr" "cidr" {
...
 resource "azurerm_subnet" "main" {
-  name                 = var.name
-  resource_group_name  = local.vnet_resource_group
-  virtual_network_name = local.vnet_name
+  name                 = var.md_metadata.name_prefix
+  resource_group_name  = local.virtual_network_resource_group_name
+  virtual_network_name = local.virtual_network_name
   address_prefixes     = [local.cidr]
-  service_endpoints    = ["Microsoft.Web"]
+  service_endpoints    = ["Microsoft.Web", "Microsoft.Storage"]
```
Comment (WillBeebe) [inline massdriver-application-azure-app-service/subnet.tf:56]: Might have something to do with blobs for app-service but I'm not sure, I've been using mostly functions recently. I don't think this hurts anything and we need a subnet module anyway.

File: massdriver-application-azure-app-service/main.tf [hunk 2]
```diff
@@ -56,7 +67,7 @@ resource "azurerm_linux_web_app" "main" {
   site_config {
     always_on                               = true
     auto_heal_enabled                       = true
-    health_check_path                       = "/health"
+    health_check_path                       = var.health_check.path
```
Comment (WillBeebe) [inline massdriver-application-azure-app-service/main.tf:70]: fix

File: massdriver-application-azure-function-app/main.tf [hunk 3]
```diff
@@ -4,12 +4,20 @@ locals {
...
   application_identity_id = azurerm_linux_function_app.main.identity[0].principal_id
   # We aren't creating an application identity for this module because we are assigning permissions directly to the system-assigned managed identity of the function app.
   create_application_identity = false
+  # The permission-assignment goes like this
+  # Azure makes the function
+  # MDXC tries to assign the role
+  # The storage account is not ready yet
+  # Without this depends_on we get intermitent, hard to debug failures.
+  depends_on = [
+    azurerm_storage_account.main
```
Comment (WillBeebe) [inline massdriver-application-azure-function-app/main.tf:19]: 🍌 s fix holy cow. I've seen this a few times but was finally able to narrow it down.

```
│ Error: loading Role Definition List: authorization.RoleDefinitionsClient#List: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ResourceNotFound" Message="The Resource 'Microsoft.Storage/storageAccounts/localdevsto000' under resource group 'local-dev-sto-000' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix"
│
│   with module.application.module.application.mdxc_application_permission.main[".connections.storage.data.security.iam.\"read/write\""],
│   on .terraform/modules/application.application/massdriver-application/main.tf line 64, in resource "mdxc_application_permission" "main":
│   64: resource "mdxc_application_permission" "main" {
│
```

I've learned to not trust implicit dependencies from the `azurerm` provider and I didn't even know that was possible, is that possible?

**Comment by WillBeebe** [comment]
I'm going to provision these both locally, brb.

File: massdriver-application-azure-app-service/_variables.tf [hunk 4]
```diff
@@ -9,6 +9,13 @@ variable "dns" {
   type = any
 }
 
+variable "health_check" {
```
Comment (xpositivityx) [inline massdriver-application-azure-app-service/_variables.tf:12]: Will this cause BC problems?

**Review by xpositivityx — APPROVED** [review]

**Review by coryodaniel — APPROVED** [review]

**Pull request merged**